### PR TITLE
Add architecture document

### PR DESCRIPTION
**Pull request by f-meloni**

### CODE REVIEW

File: docs/tutorials/architecture.html.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+**Step 1: CI**. Danger JS needs to figure out what CI we're running on. You can see them all [in
+`source/ci_source/providers`][provs]. These use ENV VARs to figure out which CI `danger ci` is running on and validate
+whether it is a pull request.
+
+**Step 2: Platform**. Next, Danger JS needs to know which platform the code review is happening in. Today it's GitHub, BitBucket Server and GitLab.
+
+**Step 3: JSON DSL**. All the metadata from the above two steps are transformed into a JSON file, which is passed into
+Danger Kotlin.
+
+**Step 4: Kotlin Plugin Setup**. Danger has to prepare your code to be compiled, so any plugins need to be set up before
```
Comment (gianluz) [inline docs/tutorials/architecture.html.md]: do you want me to add something here in this section?

File: docs/tutorials/architecture.html.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+**Step 6: Passing Results**. The results from the evaluation are turned into JSON, and then passed back to Danger JS.
+
+**Step 6: Feedback**. Danger JS reads the results, then chooses whether to create/delete/edit any messages in the code
+review site. From the results of the run, Danger JS will then pass or fail the build.
+
+[provs]: https://github.com/danger/danger-js/tree/master/source/ci_source/providers
+[dangerdsl]: https://github.com/danger/danger-js/blob/master/sourformace/dsl/DangerDSL.ts
+[runner]: https://github.com/danger/danger-js/blob/master/source/commands/danger-runner.ts
+[in_runner]: https://github.com/danger/danger-js/blob/master/source/runner/runners/inline.ts
```
Comment (gianluz) [inline docs/tutorials/architecture.html.md]: what about these references? danger js is right?

File: docs/tutorials/architecture.html.md (hunk 1 above)
Comment (f-meloni) [reply docs/tutorials/architecture.html.md]: Yes, but we also need something with more details like https://github.com/danger/swift/blob/master/Documentation/usage/extending_danger.html.md and  https://github.com/danger/swift/blob/master/Documentation/usage/extending_danger_two.html.md

File: docs/tutorials/architecture.html.md (hunk 2 above)
Comment (f-meloni) [reply docs/tutorials/architecture.html.md]: Yes, but we don't use all of them, let me remove the not used ones

**Pull request merged**